typing/typecore.ml

```diff
@@ -7356,6 +7368,35 @@ and type_expect_
             exp_type = exp2.exp_type;
             exp_attributes = sexp.pexp_attributes;
             exp_env = env }
+  | Pexp_quotation exp ->
+      let expected_mode' = expected_mode in
```

Comment:
There is no reason for this renaming.

Reply:
Thank you! My knowledge here is not great, so I appreciate the suggestion.